block_coupon.php

```diff
@@ -37,11 +37,14 @@
  * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 class block_coupon extends block_base {
+    private mixed $version;
+    private ?object $course;
```

Comment:
I know that it got assigned in the ``init()`` method.
But how do you know that it call in the constructor? The the future it may not being call at all in the parent class constructor.

To avoid this error in the future, please add the default value to the properties when you aren't assigning them in the class constructor.

```suggestion
    private mixed $version = null;
    private ?object $course = null;
```